packages/file-types/zarr/src/anndata-loaders/ObsFeatureMatrixAnndataLoader.js

```diff
@@ -367,32 +369,14 @@ export default class ObsFeatureMatrixAnndataLoader extends AbstractTwoStepLoader
     return filteredGeneNames;
   }
 
-  /* load() {
-    return Promise.all([this.loadAttrs(), this.loadCellXGene()]).then(
-      async (d) => {
-        const [{ data: attrs }, cellXGene] = d;
-        const {
-          initialFeatureFilterPath: matrixGeneFilterZarr,
-        } = this.getOptions();
-        // In order to return the correct gene list with the heatmap data,
-        // we need to filter the columns of attrs so it matches the cellXGene data.
-        if (matrixGeneFilterZarr) {
-          const matrixGeneFilter = await this.dataSource.getFlatArrDecompressed(
-            matrixGeneFilterZarr,
-          );
-          attrs.cols = attrs.cols.filter((_, i) => matrixGeneFilter[i]);
-        }
-        return Promise.resolve(new LoaderResult([attrs, cellXGene], null));
-      },
-    );
-  } */
   async load() {
+    const { path } = this.getOptions();
     const superResult = await super.load().catch(reason => Promise.resolve(reason));
     if (superResult instanceof AbstractLoaderError) {
       return Promise.reject(superResult);
     }
     return Promise.all([
-      this.dataSource.loadObsIndex(),
+      this.dataSource.loadObsIndex(path),
       this.loadInitialFilteredGeneNames(),
       this.loadCellXGene(),
```

Comment:
So related to the below, if you load the RNA `X` portion of a `mudata` object, won't this load more obs rows in the cellxgene matrix than the actual index, which is shared?